File: frontend/src/types/incident.ts
export interface Incident {
  id: string;
  project_id: string;
  title: string;
  description: string;
  severity: string;
  status: string;
  detected_at: string;
  resolved_at: string | null;
}

export interface IncidentSummary {
  id: string;
  title: string;
  severity: string;
  status: string;
  detected_at: string;
}

export interface IncidentListResponse {
  items: IncidentSummary[];
  count: number;
}

File: backend/app/schemas/incident.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class IncidentSummary(BaseModel):
    id: UUID
    title: str
    severity: str
    status: str
    detected_at: datetime

    model_config = ConfigDict(from_attributes=True)


class IncidentListResponse(BaseModel):
    items: list[IncidentSummary]
    count: int


class IncidentDetail(BaseModel):
    id: UUID
    project_id: UUID
    title: str
    description: str
    severity: str
    status: str
    detected_at: datetime
    resolved_at: datetime | None

    model_config = ConfigDict(from_attributes=True)


class IncidentStatusResponse(BaseModel):
    id: UUID
    status: str
    resolved_at: datetime | None

    model_config = ConfigDict(from_attributes=True)